lib/TimeStamp/src/timestamp.c

```diff
@@ -0,0 +1,42 @@
+#include <stm32_ll_rtc.h>
+#include <stm32_ll_rcc.h>
+#include <stm32_ll_pwr.h>
+#include "timestamp.h"
+
+void setup_RTC(void){
+    //printk("\nMade it Here\n");
+    //int async_pre = 127;
+    //int sync_pre = 311;
+    int async_pre = 124;
+    //int sync_pre = 319;
+    //Value for Renode
+    int sync_pre = 170;
+
+    LL_PWR_DisableBkUpAccess();
+    LL_RCC_SetRTCClockSource(LL_RCC_RTC_CLKSOURCE_LSI);
+    LL_RCC_EnableRTC();
+
+    LL_RTC_DisableWriteProtection(RTC);
+    LL_RTC_EnableInitMode(RTC);
+    while(!LL_RTC_IsActiveFlag_INIT(RTC)){};
```

Comment:
For these while loops, do they run for a while? If so, it would be better to put this in its own thread or have a sleep in the while statement to let other processes run.

Reply:
I think we should just add a sleep in here, probably k_msleep(500);.

Reply:
gotcha I see a situation where threads can be held up by this